Enhance Inventory Management System: New Features and Improvements

Mahmoud-B-Miqdad opened the pull request.

Mahmoud-B-Miqdad committed:
Added functionality to add a product to the inventory

Mahmoud-B-Miqdad committed:
Implemented functionality to display all products

Mahmoud-B-Miqdad committed:
Added functionality to edit a product's details

Mahmoud-B-Miqdad committed:
Implemented functionality to delete a product from the inventory

Mahmoud-B-Miqdad committed:
Added functionality to search for a product in the inventory

RandaZraik commented:
`.vs`, `bin` and `obj` folders do not have to be pushed to the repo. Please add `.gitignore` file to ignore such folders :)

RandaZraik commented on Domain/Product_Class/Product.cs:
```diff
@@ -0,0 +1,21 @@
+using System.Collections.Generic;
+using System.Linq;
+
+namespace InventoryManagementSystem.Domain.Product_Class
```
NIT: Usually, folder names for not have underscores. Also, we need that these are classes, so no need to add that to the folder name :D you can just name the folder `Models`

RandaZraik commented on Domain/Inventory_Class/Inventory.cs:
```diff
@@ -0,0 +1,114 @@
+using System.Collections.Generic;
+using System.Linq;
+
+namespace InventoryManagementSystem.Domain.Inventory_Class
```
Same comment about `Product_Class`, you can name this one `Inventory`

RandaZraik commented on Domain/General/Currency.cs:
```diff
@@ -0,0 +1,9 @@
+{
+    public enum Currency
+    {
+        Dollar,
```
A good practice is to assign an integer to each enum value.

i.e.

```csharp
Dollar = 0,
Euro = 1,
..
```

By default, values are zero-indexed, and values auto given integers starting from zero if we don't specify them. If someone adds a new value in the middle, it will get shift the numbers for the rest of values and mess up the numbers for the values.

RandaZraik commented on Domain/Inventory_Class/Inventory.cs:
```diff
@@ -0,0 +1,114 @@
+{
+    public class Inventory
+    {
+        private List<Product> products = new List<Product>();
```
NIT: Usually private properties start with `_`. i.e. `_products`

RandaZraik commented on Domain/Inventory_Class/Inventory.cs:
```diff
@@ -0,0 +1,114 @@
+            if (product == null)
+            {
+                Console.WriteLine("Product not found!");
+                Console.WriteLine("Press enter key to Back. \r\n");
```
the methods in `Inventory` class should just do direct CRUD and update products lists. If we want the user to input anything or show a message to the user for any info, this should be out of the inventory class, should be the responsibility of the caller. i.e. , `EditProduct` should receive all info that need updated and the method should update the product directly.

RandaZraik replied:
lets do this change to all methods in this class

RandaZraik commented:
@Mahmoud-B-Miqdad Nice work! thank you!

Mahmoud-B-Miqdad committed:
Update .gitignore to ignore .vs, bin, and obj folders

Mahmoud-B-Miqdad committed:
Remove ignored files from tracking

Mahmoud-B-Miqdad committed:
Moved Product and Inventory classes to Models folder

Mahmoud-B-Miqdad committed:
Updated enum values and renamed private properties

Mahmoud-B-Miqdad committed:
Refactor Inventory and Utilities classes to separate concerns and improve code structure:
- Moved user input handling to Utilities class.
- Updated Inventory class methods to perform direct CRUD operations.
- Removed user interaction from Inventory class methods.

Mahmoud-B-Miqdad commented:
@RandaZraik I've updated the project based on your feedback:
- Added a `.gitignore` file to ignore `.vs`, `bin`, and `obj` folders.
- Moved `Product` and `Inventory` classes to `Models` folder.
- Added integer assignment for `enums`.
- Used `_` prefix for private fields.
- Moved user interactions out of the `Inventory` class.

RandaZraik commented on Domain/Models/Inventory.cs:
```diff
@@ -0,0 +1,71 @@
+using System.Collections.Generic;
+using System.Linq;
+
+namespace InventoryManagementSystem.Domain.Models
```
`Models` usually used to have the DTOs/entities. Inventory can go to a different folder `Services`

Mahmoud-B-Miqdad committed:
Refactored ViewProducts to move user interaction logic from Inventory to Utilities—realized this was missed earlier, now fixed!

Mahmoud-B-Miqdad committed:
Implemented changes based on review feedback: moved Inventory class to Services folder.
